Windows EMD tests fail with NameError

Error shown [here](https://github.com/j-s-ashley/EnergyFlow/actions/runs/9858574406/job/27220276111#step:6:4121), `name '_X0' is not defined`, results from `test_n_jobs` function on [line 67](https://github.com/thaler-lab/EnergyFlow/blob/master/energyflow/tests/test_emd.py#L67).

`functools.partial(_emd4map(_X0, _X1)` results in `TypeError`.

<details>
<summary>Full error</summary>
<code>
_________________ test_n_jobs[pot-2-5-True-0.7071067811865476] _________________
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
TypeError: _emd4map() missing 1 required positional argument: 'x'
"""

The above exception was the direct cause of the following exception:

backend = 'pot', n_jobs = 2, M = 5, norm = True, R = 0.7071067811865476

    @pytest.mark.emd
    @pytest.mark.parametrize('R', [np.sqrt(2)/2, 1.0, 2])
    @pytest.mark.parametrize('norm', [True, False])
    @pytest.mark.parametrize('n_jobs, M', itertools.product([1,2,None], [5,25]))
    @pytest.mark.parametrize('backend', ['wasserstein', 'pot'])
    def test_n_jobs(backend, n_jobs, M, norm, R):
        emd, emds = (ef.emd.emd, ef.emd.emds) if backend == 'wasserstein' else (ef.emd.emd_pot, ef.emd.emds_pot)
        events = np.random.rand(nev, M, 3)
        emds1 = np.zeros((nev, nev))
        for i,ev1 in enumerate(events):
            for j in range(i):
                emds1[i,j] = emd(ev1, events[j], R=R, norm=norm)
        emds1 += emds1.T
>       emds2 = emds(events, R=R, norm=norm, verbose=0, n_jobs=n_jobs)

energyflow/tests/test_emd.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/energyflow/emd.py:895: in emds_pot
    results = pool.map(partial(_emd4map,(_X0, _X1)), local_map_args, chunksize=chunksize)
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/pool.py:367: in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.pool.MapResult object at 0x16aecdbd0>, timeout = None

    def get(self, timeout=None):
        self.wait(timeout)
        if not self.ready():
            raise TimeoutError
        if self._success:
            return self._value
        else:
>           raise self._value
E           TypeError: _emd4map() missing 1 required positional argument: 'x'

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/pool.py:774: TypeError
</code>
</details>

--- a/energyflow/emd.py
+++ b/energyflow/emd.py
@@ -835,7 +835,6 @@ def emds_pot(X0, X1=None, R=1.0, norm=False, beta=1.0, measure='euclidean', coor
     phi_col_m1 = phi_col - 1
 
     # process events into convenient form for EMD
-    global _X0, _X1
     start = time.time()
     args = (norm, gdim, periodic_phi, phi_col_m1,
             mask, R, hadr2cart, euclidean, error_on_empty)
@@ -918,9 +917,6 @@ def emds_pot(X0, X1=None, R=1.0, norm=False, beta=1.0, measure='euclidean', coor
     # unrecognized n_jobs value
     else:
         raise ValueError('n_jobs must be a positive integer or -1')
-
-    # delete global arrays
-    del _X0, _X1
 
     # if doing an array with itself, symmetrize the distance matrix
     if sym:

--- a/energyflow/emd.py
+++ b/energyflow/emd.py
@@ -519,7 +519,7 @@ def _cdist(X, Y, euclidean, periodic_phi, phi_col):
     return out
 
 # helper function for pool imap
-def _emd4map(x):
+def _emd4map(_X0, _X1, x):
     (i, j), params = x
     return _emd(_X0[i], _X1[j], *params)
 

--- a/energyflow/emd.py
+++ b/energyflow/emd.py
@@ -891,7 +891,7 @@ def emds_pot(X0, X1=None, R=1.0, norm=False, beta=1.0, measure='euclidean', coor
                 local_map_args = [next(map_args) for i in range(end - begin)]
 
                 # map function and store results
-                results = pool.map(_emd4map, local_map_args, chunksize=chunksize)
+                results = pool.map(functools.partial(_emd4map(_X0, _X1)), local_map_args, chunksize=chunksize)
                 for arg,r in zip(local_map_args, results):
                     i, j = arg[0]
                     emds[i, j] = r

--- a/energyflow/emd.py
+++ b/energyflow/emd.py
@@ -45,6 +45,7 @@ are used instead of energies.
 # Copyright (C) 2017-2022 Patrick T. Komiske III and Eric Metodiev
 
 from __future__ import absolute_import, division, print_function
+from functools import partial
 
 import itertools
 import multiprocessing
@@ -891,7 +892,7 @@ def emds_pot(X0, X1=None, R=1.0, norm=False, beta=1.0, measure='euclidean', coor
                 local_map_args = [next(map_args) for i in range(end - begin)]
 
                 # map function and store results
-                results = pool.map(functools.partial(_emd4map(_X0, _X1)), local_map_args, chunksize=chunksize)
+                results = pool.map(partial(_emd4map,_X0, _X1), local_map_args, chunksize=chunksize)
                 for arg,r in zip(local_map_args, results):
                     i, j = arg[0]
                     emds[i, j] = r

